ts/src/bullish.ts

```diff
@@ -0,0 +1,2404 @@
+        //                     },
+        //                     {
+        //                         "feeTierId": "3",
+        //                         "staticSpreadFee": "0.00010000",
+        //                         "isDislocationEnabled": false
+        //                     },
+        //                     {
+        //                         "feeTierId": "4",
+        //                         "staticSpreadFee": "0.00010000",
+        //                         "isDislocationEnabled": true
+        //                     },
+        //                     {
+        //                         "feeTierId": "5",
+        //                         "staticSpreadFee": "0.00020000",
+        //                         "isDislocationEnabled": false
+        //                     },
+        //                     {
+        //                         "feeTierId": "6",
+        //                         "staticSpreadFee": "0.00020000",
+        //                         "isDislocationEnabled": true
+        //                     },
+        //                     {
+        //                         "feeTierId": "7",
+        //                         "staticSpreadFee": "0.00060000",
+        //                         "isDislocationEnabled": false
+        //                     },
+        //                     {
+        //                         "feeTierId": "8",
+        //                         "staticSpreadFee": "0.00060000",
+        //                         "isDislocationEnabled": true
+        //                     },
+        //                     {
+        //                         "feeTierId": "9",
+        //                         "staticSpreadFee": "0.00100000",
+        //                         "isDislocationEnabled": false
+        //                     }
+        //                 ],
+        //             "marketType": "DATED_FUTURE",
+        //             "contractMultiplier": "1",
+        //             "settlementAssetSymbol": "USDC",
+        //             "underlyingQuoteSymbol": "USDC",
+        //             "underlyingBaseSymbol": "BTC",
+        //             "openInterestLimitUSD": "100000000.0000",
+        //             "concentrationRiskPercentage": "100.00",
+        //             "concentrationRiskThresholdUSD": "30000000.0000",
+        //             "expiryDatetime": "2025-05-16T08:00:00.000Z",
+        //             "priceBuffer": "0.1",
+        //             "feeGroupId": "4"
+        //         },
+        //     ]...
+        //
+        return this.parseMarkets (response);
+    }
+
+    parseMarket (market: Dict): Market {
+        const id = this.safeString (market, 'symbol');
+        let symbol = undefined;
+        const baseId = this.safeString (market, 'baseSymbol');
+        const quoteId = this.safeString (market, 'quoteSymbol');
+        const base = this.safeString (market, 'baseSymbol');
```

Comment:
base means the ccxt-unified base, we should convert it using safeCurrencyCode, we should not read it directly from the exchange market (the same applies to quote/settle/etc)